### PR TITLE
Closes #2960 int version of memory_usage

### LINKED ISSUE #2960: int version of memory_usage
Create a version of DataFrame.memory_usage that returns the memory as a int rather than a string.

For example:
'''
df.memory_usage(unit="GB")
'''
returns an integer.

### PR DESCRIPTION
**Pull request by ajpotts**
Closes #2960 int version of memory_usage

Creates the following new functions to characterize the memory usage of various objects:
pdarray.nbytes
Index.memory_usage
MultiIndex.memory_usage
Series.memory_usage
DataFrame.memory_usage
DataFrame.memory_usage_info

### CODE REVIEW

File: arkouda/index.py [hunk 1]
```diff
@@ -192,6 +192,41 @@ def from_return_msg(cls, rep_msg):
 
         return cls.factory(idx) if len(idx) > 1 else cls.factory(idx[0])
 
+    def memory_usage(self, unit="B"):
+        """
+        Return the memory usage of the Index values.
+
+        Parameters
+        ----------
+        unit : str, default = "GB"
```
[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/index.py:201]: I set the default to match pandas:
https://pandas.pydata.org/docs/reference/api/pandas.Index.memory_usage.html

However, I'm open to changing it.

[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/index.py]: OK, I see it was a doc string bug.  It should be fixed to match now.  I'm leaving the default to "B".

File: PROTO_tests/tests/index_test.py [hunk 2]
```diff
@@ -50,6 +50,27 @@ def test_multiindex_creation(self, size):
...
             idx = ak.MultiIndex([ak.arange(size), ak.arange(size - 1)])
 
+    def test_memory_usage(self):
+        from arkouda.dtypes import BigInt
+        from arkouda.index import Index, MultiIndex
+
+        idx = Index(ak.cast(ak.array([1, 2, 3]), dt="bigint"))
+        assert idx.memory_usage() == 3 * BigInt.itemsize
+
+        n = 2000
```
Comment (jaketrookman) [inline PROTO_tests/tests/index_test.py:60]: Everything looks good, I would just recommend changing this to parameterizing `n` using the `@pytest.mark.parametrize("size", pytest.prob_size)` that other PROTO tests use

Comment (ajpotts) [reply PROTO_tests/tests/index_test.py]: I made the changes.  Let me know what you think.

**Review by jaketrookman — APPROVED** [review]
Looks great!

**Review by stress-tess — APPROVED** [review]
looks good to me!

**Pull request merged**